lib/matplotlib/axes/_base.py

```diff
@@ -2208,46 +2208,44 @@ def update_datalim_bounds(self, bounds):
         """
         self.dataLim.set(mtransforms.Bbox.union([self.dataLim, bounds]))
 
-    def _process_unit_info(self, xdata=None, ydata=None, kwargs=None):
+    def _process_unit_info(self, datasets=None, kwargs=None):
         """
-        Look for unit *kwargs* and update the axis instances as necessary
+        Set axis units based on *datasets* and *kwargs*.
 
-
-        .. warning ::
-
-           This method may mutate the dictionary passed in an kwargs and
-           the Axis instances attached to this Axes.
-        """
-
-        def _process_single_axis(data, axis, unit_name, kwargs):
-            # Return if there's no axis set
+        Parameters
+        ----------
+        datasets : dict
+            Mapping of axis names (per `._get_axis_map`) to datasets.
+        kwargs : dict
+            Other parameters from which unit info may be popped.  Note that
```

Comment:
This is a bit vague. Can we be more explicit? Which entries will be popped under which conditions?

Reply:
done